test/common/crudtester.go

```diff
@@ -362,6 +365,63 @@ func (c *FederatedTypeCrudTester) CheckDelete(fedObject *unstructured.Unstructur
 	}
 }
 
+func (c *FederatedTypeCrudTester) SetDeleteOption(fedObject *unstructured.Unstructured, opts ...client.DeleteOption) {
+	apiResource := c.typeConfig.GetFederatedType()
+	qualifiedName := util.NewQualifiedName(fedObject)
+	kind := apiResource.Kind
+	_, err := c.updateObject(apiResource, fedObject, func(obj *unstructured.Unstructured) {
+		util.ApplyDeleteOptions(obj, opts...)
+	})
+	if err != nil {
+		c.tl.Fatalf("Error updating %s %q: %v", kind, qualifiedName, err)
+	}
+}
+
+func (c *FederatedTypeCrudTester) CheckReplicaSet(fedObject *unstructured.Unstructured) {
+	lb, ok, _ := unstructured.NestedStringMap(fedObject.Object, "spec", "selector", "matchLabels")
+	if !ok {
+		c.tl.Fatal("Failed to get matchLabels on the target deployment")
+	}
+
+	matchingLabels := (client.MatchingLabels)(lb)
+
+	for clusterName := range c.testClusters {
+		clusterConfig := c.testClusters[clusterName].Config
+
+		kubeClient := kubeclientset.NewForConfigOrDie(clusterConfig)
+		WaitForNamespaceOrDie(c.tl, kubeClient, clusterName, fedObject.GetNamespace(),
+			c.waitInterval, 30*time.Second)
```

Comment:
please use `framework.PollInterval, framework.TestContext.SingleCallTimeout` instead of `c.waitInterval, 30*time.Second`.

Reply:
I guess the `framework` package depends on `common`,  so it doesn't allow me to import `framework` in crudtester.go. 
Any idea to resolve this? I'm pretty new to Golang.

Reply:
I believe you faced some cycle go deps here. It is alright, I'll move those common attributes to a different package. Not a blocker.